core/vm/eips.go

```diff
@@ -241,3 +242,32 @@ func enable3860(jt *JumpTable) {
 	jt[CREATE].dynamicGas = gasCreateEip3860
 	jt[CREATE2].dynamicGas = gasCreate2Eip3860
 }
+
+func enable5656(jt *JumpTable) {
+	jt[MCOPY] = &operation{
+		execute:     opMcopy,
+		constantGas: GasFastestStep,
+		dynamicGas:  gasMemoryCopy,
+		minStack:    minStack(3, 0),
+		maxStack:    maxStack(3, 0),
+		memorySize:  memoryMemoryCopy,
+	}
+}
+
+func opMcopy(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]byte, error) {
+	var (
+		dst    = scope.Stack.pop()
+		src    = scope.Stack.pop()
+		length = scope.Stack.pop()
+	)
+	src, overflow := src.Uint64WithOverflow()
```

Comment:
Aren't both `src` and `dst` equally checked for overflow during gas cost calc?  It does check that the 'largest' does not overflow, doesn't it?

Reply:
addressed in ee16ae9a0efd784bbbb90be6d343ecdf98e8ea69